src/pages/index.js

```diff
@@ -4,128 +4,187 @@ import FormValidator from "../components/FormValidator.js";
-// const config = {
-//   formSelector: ".modal__form",
-//   inputSelector: ".modal__input",
-//   submitButtonSelector: ".modal__button",
-//   inactiveButtonClass: "modal__button_disabled",
-//   inputErrorClass: "modal__input_type_error",
-//   errorClass: "modal__error_visible",
-// };
+const editAvatarButton = document.querySelector(".profile__edit-avatar-button");
+
+// const editAvatarModal = document.querySelector("#edit-avatar-modal");
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "5f364a70-6354-4d7f-b3e1-21a8fcface2d",
+    "Content-Type": "application/json",
+  },
+});
 const editProfileValidator = new FormValidator(config, "#profile-edit-form");
 const addCardValidator = new FormValidator(config, "#add-card-form");
-
-const cardSection = new Section(
-  { items: initialCards, renderer: createCard },
-  ".cards__list"
-);
+const editAvatarValidator = new FormValidator(config, "#edit-avatar-form");
 
 const editProfilePopup = new PopupWithForm(
   "#profile-edit-modal",
   handleProfileEditSubmit
 );
 
 const addCardPopup = new PopupWithForm("#add-card-modal", handleAddCardSubmit);
+const editAvatarPopup = new PopupWithForm(
+  "#edit-avatar-modal",
+  handleEditAvatarSubmit
+);
 const previewImagePopup = new PopupWithImage("#preview-image-modal");
+const deleteConfirmationPopup = new PopupWithConfirmation(
+  "#delete-confirmation-modal"
+);
 const profileUserInfo = new UserInfo({
   nameSelector: ".profile__title",
   jobSelector: ".profile__description",
+  avatarSelector: ".profile__image",
 });
 
+/* Event Handlers */
+function handleEditAvatarSubmit(inputValues) {
+  editAvatarPopup.renderLoadingSave(true);
+  api
+    .updateAvatar(inputValues.link)
+    .then((res) => {
+      profileUserInfo.setUserAvatar(res.avatar);
+      editAvatarPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+      alert(`${err}. Failed to edit avatar :(`); // log the error to the console
+    })
+    .finally(() => {
+      editAvatarPopup.renderLoadingSave(false);
+    });
+}
+
 function handleImageClick(data) {
   previewImagePopup.open(data.name, data.link);
 }
 
-/* Event Handlers */
 function handleProfileEditSubmit(inputValues) {
+  editProfilePopup.renderLoadingSave(true);
   //should be handled by userinfo class
-  profileUserInfo.setUserInfo(inputValues.title, inputValues.description);
+
+  api
+    .updateProfileInfo(inputValues) //{title: ddd, description: kk}
+    .then((data) => {
+      profileUserInfo.setUserInfo(inputValues.title, inputValues.description);
+      editProfilePopup.close();
+      //profileUserInfo.setUserInfo(data.name, data.about);
+      //console.log("Name:", data.name, "Job:", data.about);
+    })
+    .catch((err) => {
+      console.error(err);
+      alert(`${err}. Failed to update profile info :(`); // log the error to the console
+    })
+    .finally(() => {
+      editProfilePopup.renderLoadingSave(false);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```